services/motion/builtin/builtin_test.go

```diff
@@ -1731,25 +1279,897 @@ func TestMoveOnGlobe(t *testing.T) {
+		defer ms.Close(ctx)
+		msBuiltin, ok := ms.(*builtIn)
+		test.That(t, ok, test.ShouldBeTrue)
+
+		// Overwrite `logger` so we can use a handler to test for correct log messages
+		logger, observer := logging.NewObservedTestLogger(t)
+		msBuiltin.logger = logger
+
+		req := motion.MoveOnMapReq{
+			ComponentName: base.Named("test-base"),
+			Destination:   goal1SLAMFrame,
+			SlamName:      slam.Named("test_slam"),
+			Extra:         map[string]interface{}{"smooth_iter": 5},
+		}
+
+		executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+
+		timeoutCtx, timeoutFn := context.WithTimeout(ctx, time.Second*5)
+		defer timeoutFn()
+		err = motion.PollHistoryUntilSuccessOrError(timeoutCtx, ms, time.Millisecond*5, motion.PlanHistoryReq{
+			ComponentName: req.ComponentName,
+			ExecutionID:   executionID,
+			LastPlanOnly:  true,
+		})
+		test.That(t, err, test.ShouldBeNil)
+
+		endPos, err := kb.CurrentPosition(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		logger.Debug(spatialmath.PoseToProtobuf(endPos.Pose()))
+		test.That(t, spatialmath.PoseAlmostEqualEps(endPos.Pose(), goal1BaseFrame, 10), test.ShouldBeTrue)
 
-	executionID, err := ms.MoveOnMapNew(ctx, req)
-	test.That(t, err.Error(), test.ShouldEqual, "unimplemented")
-	test.That(t, executionID, test.ShouldResemble, uuid.Nil)
+		// Now, we try to go to the second goal. Since the `CurrentPosition` of our base is at `goal1`, the pose that motion solves for and
+		// logs should be {x:-1043  y:593}
+		req = motion.MoveOnMapReq{
+			ComponentName: base.Named("test-base"),
+			Destination:   goal2SLAMFrame,
+			SlamName:      slam.Named("test_slam"),
+			Extra:         map[string]interface{}{"smooth_iter": 5},
+		}
+		executionID, err = ms.(*builtIn).MoveOnMapNew(context.Background(), req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+
+		timeoutCtx, timeoutFn = context.WithTimeout(ctx, time.Second*5)
+		defer timeoutFn()
+		err = motion.PollHistoryUntilSuccessOrError(timeoutCtx, ms, time.Millisecond*5, motion.PlanHistoryReq{
+			ComponentName: req.ComponentName,
+			ExecutionID:   executionID,
+			LastPlanOnly:  true,
+		})
+		test.That(t, err, test.ShouldBeNil)
+
+		endPos, err = kb.CurrentPosition(ctx)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, spatialmath.PoseAlmostEqualEps(endPos.Pose(), goal2BaseFrame, 5), test.ShouldBeTrue)
+
+		// We don't actually surface the internal motion planning goal; we report to the user in terms of what the user provided us.
+		// Thus, we must do string surgery on the internal `motionplan` logs to extract the requested relative pose and check it is correct.
+		goalLogsObserver := observer.FilterMessageSnippet("Goal: reference_frame:").All()
+		test.That(t, len(goalLogsObserver), test.ShouldEqual, 2)
+		logLineToGoalPose := func(logString string) spatialmath.Pose {
```

Comment:
Is there really no way to test this other than to parse the logs which are logged during the test?

Reply:
I kinda think this test should exist somewhere else, which has access to these internals rather than smuggled out through the logger.

Reply:
Could we not test this by calling `PlanHistory`?

Reply:
i think so, looking into this right now 👀